src/java.base/share/native/libcrexec/image_score.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2025, Azul Systems, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+#ifndef IMAGE_SCORE_HPP
+#define IMAGE_SCORE_HPP
+
+#include <cstring>
+#include <cstdlib>
+#include <cstdint>
+
+#include "crlib/crlib_image_constraints.h"
+#include "linkedlist.hpp"
+
+class ImageScore {
+private:
+
+  struct Score {
+    const char* _name;
+    double _value;
+
+    Score(const char* name, double value): _name(name), _value(value) {}
+
+    Score(Score &&o) {
+      _name = o._name;
+      _value = o._value;
+      o._name = nullptr;
+    }
+
+    ~Score() {
+      free((void*) _name);
+    }
+  };
+
+  LinkedList<Score> _score;
+
+public:
+  bool set_score(const char* name, double value) {
+    const char* name_copy = strdup(name);
+    if (name_copy == nullptr) {
+      return false;
+    }
+    return _score.add(Score(name_copy, value));
+  }
```

Comment:
This does not adhere to the engine API specification that when a metric name is repeated the old value is replaced